### PR TITLE
Add doc: sync module with the latest template AND pre-commit fix

### LINKED ISSUE #32: Sync template with modules
## Problem
Once a user creates a new repository from the template, that new repository has no common git-history with the original template. 

## How to resolve
Originally posted [here](https://github.com/fredrikhgrelland/terraform-nomad-minio/pull/9#issue-476234355)

Basically  **git merge**. 

### Applied strategy - the first time only
1. Add remote upstream to template origin `git remote add template [URL of the template repo]`
2. Fetch all `git fetch --all`
3. Checkout **template** master branch. Its important to have it locally. 
4. Checkout new branch from **origin** master
5. Run git merge with the flag `--allow-unrelated-histories` 
6. fix conflicts
7. commit
8. PR against current master

[ref stackoverflow](https://stackoverflow.com/a/56577320)

### LINKED ISSUE #41: Hotfix: pre-commit need to check for terraform binary
Use the `check_for_terraform_binary` target

### PR DESCRIPTION
**Pull request by zhenik**

### CODE REVIEW

**Review by dangernil — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -445,6 +446,34 @@ All PRs will run [super-linter](https://github.com/github/super-linter). You can
...
+git checkout -b sync master
+git pull
+```
+5 Run git merge with the flag --allow-unrelated-histories
+```text
+git merge template-master --allow-unrelated-histories
+```
+6 Fix conflicts (carefully)  
+7 Commit changes  
+8 Make pull request
```
Comment (Neha-Sinha2305) [inline README.md]: It is obvious but then good to have this step anyway.
```suggestion
8 Push branch sync
9 Make pull request `sync` against `master` branch
```

**Review by Neha-Sinha2305 — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -445,6 +446,35 @@ All PRs will run [super-linter](https://github.com/github/super-linter). You can
...
+```
+3 Checkout template master branch and pull. Its important to have it locally
+```text
+git checkout -b template-master template/master
+git pull
+```
+4 Checkout new branch from origin master of current module
+```text
+git checkout -b sync master
+git pull
```
Comment (Neha-Sinha2305) [inline README.md]: what is this pull here for? Do lines 466,467 do the same job as the below :
```
git checkout master
git pull
git checkout -b sync
```

Comment (zhenik) [reply README.md]: actually no. I think your version is correct

Comment (zhenik) [inline README.md]:
```suggestion
git checkout master    # checkout master
git pull               # pull the latest master
git checkout -b sync   # checkout new branch `sync` from master
```

Comment (zhenik) [reply README.md]: @Neha-Sinha2305 WDYT?

Comment (Neha-Sinha2305) [reply README.md]: looks good to me

**Review by Neha-Sinha2305 — APPROVED** [review]
LGTM

**Pull request merged**